### ISSUE TITLE
Using Proxy

### ISSUE THREAD

**Issue by drtiwarir** [ameshkov/sniproxy#5]
I just saw this project, was looking for something like this.
I use a socks proxy on my phone (made using [proxidize-android](https://github.com/proxidize/proxidize-android)
Was wondering if I use sniproxy and forward selected domians to socks, it will be working very similar to commercially available DNS proxies.
How can I add username & password  for sniproxy to forward requests to socks ? because socks proxy on phone needs username and password to allow connection.
Please advice.
Thanks

**Comment by ameshkov** [comment]
> Was wondering if I use sniproxy and forward selected domians to socks, it will be working very similar to commercially available DNS proxies.

It will be pretty similar but not the same of course.

> How can I add username & password for sniproxy to forward requests to socks ? because socks proxy on phone needs username and password to allow connection.

You'll need to patch sniproxy to achieve this.

**Comment by drtiwarir** [comment]
Thanks Ameshkov
I tried to patch sniproxy for adding username and password option, but not working, can you please help.
Thanks

**Comment by ameshkov** [comment]
Please open a pull request, I'll try to help

**Comment by ameshkov** [comment]
Quick question, have you tried `proto://user:password@host:password` URL?

**Comment by drtiwarir** [comment]
Nope, I tried a few things but did not work, can you please add ?
Thanking you

**Comment by ameshkov** [comment]
I checked the code and this approach with adding authentication to a proxy URL should work

**Comment by drtiwarir** [comment]
I wil try 
Thanks
